test/unit/internal/test-iterators.cpp

```diff
@@ -0,0 +1,180 @@
+  ASSERT_LT(three, five);
+  ASSERT_GE(five, three);
+  ASSERT_GT(five, three);
+  ASSERT_NE(five, three);
+  ASSERT_EQ(three + 2, five);
+  ASSERT_EQ(2 + three, five);
+  ASSERT_EQ(five - 2, three);
+  ASSERT_EQ(8 - five, three);
+}
+
+TYPED_TEST(StridedNumericIteratorTest, simple)
+{
+  RAJA::Iterators::strided_numeric_iterator<TypeParam> i(0, 2);
+  ASSERT_EQ(0, *i);
+  ++i;
+  ASSERT_EQ(2, *i);
+  --i;
+  ASSERT_EQ(0, *i);
+  i += 2;
+  ASSERT_EQ(4, *i);
+  i -= 1;
+  ASSERT_EQ(2, *i);
+  RAJA::Iterators::strided_numeric_iterator<TypeParam> three(3, 2);
+  RAJA::Iterators::strided_numeric_iterator<TypeParam> five(5, 2);
+  ASSERT_LE(three, three);
+  ASSERT_LE(three, five);
+  ASSERT_LT(three, five);
+  ASSERT_GE(five, three);
+  ASSERT_GT(five, three);
+  ASSERT_NE(five, three);
+  ASSERT_EQ(three + 1, five);
+  ASSERT_EQ(five - 1, three);
+}
+
+#if defined(ENABLE_ITERATOR_OVERFLOW_DEBUG)
+TYPED_TEST(NumericIteratorTest, overflow)
+{
+  if (std::is_unsigned<TypeParam>::value) {
+    ASSERT_ANY_THROW({
+      TypeParam val = 10;
+      RAJA::Iterators::numeric_iterator<TypeParam> of_it(val);
+      of_it -= 11;
+    });
+    ASSERT_ANY_THROW({
+      TypeParam val = std::numeric_limits<TypeParam>::max() - 10;
+      RAJA::Iterators::numeric_iterator<TypeParam> of_it(val);
+      of_it += 11;
+    });
+  
+    ASSERT_ANY_THROW({
+      TypeParam val = 10;
+      RAJA::Iterators::numeric_iterator<TypeParam> of_it(val);
+      auto sum = of_it - 11u;
+      (void)sum;
+    });
+    ASSERT_ANY_THROW({
+      TypeParam val = std::numeric_limits<TypeParam>::max() - 10;
+      RAJA::Iterators::numeric_iterator<TypeParam> of_it(val);
+      auto sum = of_it + 11;
+      (void)sum;
+    });
+  
+    ASSERT_ANY_THROW({
+      TypeParam val = 10;
+      const RAJA::Iterators::numeric_iterator<TypeParam> of_it(val);
+      auto sum = 8 - of_it;
+      (void)sum;
+    });
+    ASSERT_ANY_THROW({
+      TypeParam val = std::numeric_limits<TypeParam>::max() - 10;
+      const RAJA::Iterators::numeric_iterator<TypeParam> of_it(val);
+      auto sum = 11 + of_it;
+      (void)sum;
+    });
+  } 
+}
+
+TYPED_TEST(StridedNumericIteratorTest, overflow)
+{
+  if (std::is_unsigned<TypeParam>::value){
+    ASSERT_ANY_THROW({
+      TypeParam val = 2;
+      RAJA::Iterators::strided_numeric_iterator<TypeParam> of_it(val, 2);
+      of_it -= 2;
+    });
+    ASSERT_ANY_THROW({
+      TypeParam val = std::numeric_limits<TypeParam>::max() - 2;
+      RAJA::Iterators::strided_numeric_iterator<TypeParam> of_it(val, 2);
+      of_it += 2;
+    });
+
+    ASSERT_ANY_THROW({
+      TypeParam val = 2;
+      RAJA::Iterators::strided_numeric_iterator<TypeParam> of_it(val, 2);
+      auto sum = of_it - 2;
+      (void)sum;
+    });
+    ASSERT_ANY_THROW({
+      TypeParam val = std::numeric_limits<TypeParam>::max() - 2;
+      RAJA::Iterators::strided_numeric_iterator<TypeParam> of_it(val, 2);
+      auto sum = of_it + 2;
+      (void)sum;
+    });
+  }
+}
+#endif
+
```

Comment:
I think there are not enough newlines at the end of this file 😄

Reply:
white space is good 🥇